lightning/restore/tidb.go

```diff
@@ -280,19 +280,41 @@ func ObtainNewCollationEnabled(ctx context.Context, db *sql.DB) bool {
 }
 
 func AlterAutoIncrement(ctx context.Context, db *sql.DB, tableName string, incr int64) error {
-	sql := common.SQLWithRetry{
+	sqlDB := common.SQLWithRetry{
 		DB:     db,
 		Logger: log.With(zap.String("table", tableName), zap.Int64("auto_increment", incr)),
 	}
-	query := fmt.Sprintf("ALTER TABLE %s AUTO_INCREMENT=%d", tableName, incr)
-	task := sql.Logger.Begin(zap.InfoLevel, "alter table auto_increment")
-	err := sql.Exec(ctx, "alter table auto_increment", query)
+
+	task := sqlDB.Logger.Begin(zap.InfoLevel, "alter table auto_increment")
+	var query string
+	err := sqlDB.Transact(ctx, "alter table auto_increment", func(i context.Context, tx *sql.Tx) error {
+		fetchQuery := fmt.Sprintf("SHOW TABLE %s NEXT_ROW_ID", tableName)
+		var (
+			dbName, tblName, columnName, idType string
+			nextID                              int64
+		)
+		err := tx.QueryRowContext(ctx, fetchQuery).Scan(&dbName, &tblName, &columnName, &nextID, &idType)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		if incr > nextID {
+			nextID = incr
+		}
```

Comment:
this check is totally unnecessary because `ALTER TABLE AUTO_INCREMENT` will never decrease the auto_increment value.

Reply:
There is a scenario that if some of the files have auto id column value, but some are not. The the next auto calculated by lightning may be smaller than tidb side. But we still need to update tidb auto id to refresh all tidb node local cache, so this check is needed to avoid tidb doing nothing.

Reply:
`alter table auto_increment` never does nothing according to this comment:

https://github.com/pingcap/tidb/blob/master/ddl/ddl_api.go#L2528-L2533

Reply:
I have reset these codes to the original version since tidb will do it.